frontend/src/components/CourseModal/EvaluationResponses.tsx

```diff
@@ -164,6 +164,26 @@ const EvaluationResponses: React.FC<{
 
   return (
     <div>
+      <StyledInput
+        id="filter-input"
+        type="text"
+        placeholder="Search evaluations..."
+        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Hmm we probably shouldn't modify the elements after they have been rendered here by accessing the DOM – can you refactor this so we store a state variable for the input text and filter the arrays before rendering them below?

Reply:
How would you be able to filter the arrays before rendering them? Wouldn't that require refreshing and rewriting the DOM elements every time the filter changes? That seems far more inefficient than this - I'm not technically modifying the elements at all, just hiding/showing them

Reply:
Yes – you'd call `.filter` on the arrays. It should be efficient enough, it's what we do for the main search (see https://github.com/coursetable/coursetable/blob/master/frontend/src/contexts/searchContext.tsx)

Reply:
i.e. like this: https://gigi.nullneuron.net/gigilabs/filter-list-as-you-type-with-react/